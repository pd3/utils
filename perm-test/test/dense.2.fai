1	56
2	56
3	56
4	56
5	56
