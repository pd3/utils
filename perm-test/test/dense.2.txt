
perm-test -t test/dense.2.tgt.txt -b test/dense.2.bg.txt -f test/dense.2.fai -c test/dense.2.calls.txt -n 1e9

             10        20        30        40        50
    12345678901234567890123456789012345678901234567890123456
       bbb bbbb bbb           bbb              bbbb      bbb

    12345678901234567890123456789012345678901234567890123456
       bbb bbbb bbb           bbb              bbbb      bbb

    12345678901234567890123456789012345678901234567890123456
       bbb bbbb bbb           bbb              bbbb      bbb

    12345678901234567890123456789012345678901234567890123456
       bbb bbbb bbb           bbb              bbbb      bbb

    12345678901234567890123456789012345678901234567890123456
       bbb bbbb bbb     ttt   bbb   ttt ttt    bbbb tttt bbb
                    ccccccc
    ....... ....... ....ooo ....... ooo.ooo ....... xxxx...
     ....... ....... ...ooo. ....... oo.ooo. ....... xxx....
      ....... ....... ..ooo.. ......x ..ooo.. ......x
       ....... ....... .ooo... .....xx .ooo... .....xx
        ....... ....... xxx.... ....xx. ooo.... ....xxx
         ....... ....... xx..... ...ooo. xx..... ...xxxx
          ....... ......x x...... ..ooo.o x...... ..xxxx.
           ....... .....oo ....... .ooo.oo ....... .xxxx..

    exact: 30./(5*50) = 0.12
    perm-test:          0.120015051
    err:   1.5e-05

perm-test -t test/dense.2.tgt.txt -b test/dense.2.bg.txt -f test/dense.2.fai -c test/dense.2.calls.txt -n 1e9 --no-bg-overlap

    exact: 13./(5*50) = 0.052
    perm-test:      0.052001216
    err:   1.216e-06


