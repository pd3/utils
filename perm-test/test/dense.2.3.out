# NITER_ROUNDS:
#    - number of iterations total
#    - number of batches
# NCALLS:
#    - number of calls used
#    - number of calls skipped because of their size or inaccessibility
# TEST_ENR:
#    - number of iterations
#    - number of times the simulation had the same or more hits than observed in input data
#    - P-value derived from the two fields
# TEST_DPL:
#    - number of iterations
#    - number of times the simulation had the same or fewer hits than observed in input data
#    - P-value derived from the two fields
# TEST_FOLD:
#    - number of hits in the input data
#    - average number of hits in simulations
#    - average of standard deviations approximated from each batch using the current average estimate
# DIST, the distribution of number of hits in the simulations:
#    - numbef of hits in a simulation: the start of the half-closed, half-open distribution bin [beg,end)
#    - numbef of hits in a simulation: the end of the half-closed, half-open distributin bin [beg,end)
#    - number of simulations with this many hits (raw count)
#    - number of simulations with this many hits (density)
SEED	1577568659
NITER_ROUNDS	1.000000e+05	1
MSG	Maximum chromosome randomization error due to length discretization: 1.5e-03%
NCALLS	1	0
TEST_ENR	100000	5210	5.210000e-02
TEST_DPL	100000	100000	1.000000e+00
TEST_FOLD	1	0.052100	0.222229
DIST	0	1	94790	9.479000e+04
DIST	1	2	5210	5.210000e+03
